yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -340,6 +340,14 @@ private[spark] class Client(
           "for alternatives.")
     }
 
+    // Uploading $SPARK_CONF_DIR/log4j.properties file to the distributed cache to make sure that
+    // the executors will use the latest configurations instead of the default values. This is
+    // required when user changes log4j.properties directly to set the log configurations. If
+    // configuration file is provided through --files then executors will be taking configurations
+    // from --files instead of $SPARK_CONF_DIR/log4j.properties.
+    val log4jConf = oldLog4jConf.orElse(
+      Option(Utils.getContextOrSparkClassLoader.getResource("/log4j.properties")).map(_.toString))
```

Comment:
Sure, but we're trying to make the default case easier. If the user is messing with `log4j.configuration` then he probably knows what he's doing, and this change would not affect that.